### PR TITLE
updated c/c++ flag guidance again

### PR DESCRIPTION
**Pull request by AWSjswinney**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: c-c++.md [hunk 1]
```diff
@@ -2,12 +2,14 @@
 
 ### Enabling Arm Architecture Specific Features
 
+TLDR: To target all current generation Graviton instanes, use `-march=arm8.2-a`.
```
[the remark this answers is not in the document]
Comment (AWSjswinney) [reply c-c++.md:5]: I can never sneak those typos past you! 🤣 

By "current generation" I mean anything that isn't filtered out on the EC2 launch page when you disable "all generations". There's also a flag that says "current generation: true" for all Graviton2 and Graviton3 instances. Only  the A1 instances are "current generation: false". I see you can also get that information from `aws ec2 describe-instance-types`.

Screenshot 2024-01-23 at 11 07 20

I wonder if there is a better way to write that...

```suggestion
TLDR: To target all current generation* Graviton instances, use `-march=arm8.2-a`.

* Current generation instances as noted in the EC2 launch instances page or from the AWS API: `aws ec2 describe-instances`.
```

[the remark this answers is not in the document]
Comment (snadampal) [reply c-c++.md:5]: I agree with @AWSjswinney,  current generation is not just the latest, anything that is actively supported.

**Review by snadampal — APPROVED** [review]

File: c-c++.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lrbison) [reply c-c++.md:5]: In that case I would find "`*6g` and newer instances" more clear than "current generation", as that saves me from having to go open some aws console page and check.

**Review by lrbison — APPROVED** [review]
